pcmanfm/tabpage.cpp

```diff
@@ -465,7 +465,12 @@ void TabPage::onFolderFinishLoading() {
     // After finishing loading the folder, the model is updated, but Qt delays the UI update
     // for performance reasons. Therefore at this point the UI is not up to date.
     // For example, the scrollbar ranges are not updated yet. We solve this by installing an Qt timeout handler.
-    QTimer::singleShot(10, this, &TabPage::onUiUpdated);
+    QTimer::singleShot(10, this, [this] {
+        onUiUpdated();
+        // When the UI is fully loaded if there are files to select we select them
+        if(!filesToSelect_.isEmpty())
+            onSelectItems();
+    });
```

Comment:
There's no need to this change and to the new method `onSelectItems()`. Instead, you could add the contents of `onSelectItems()` to the end of `onUiUpdated()` (with an appropriate comment) and remove it. I'll review the contents later (it seems you've changed it).